### PR TITLE
Chord

### PR DESCRIPTION
**Pull request by HeikoBornholdt**

### CODE REVIEW

**Comment by HeikoBornholdt** [comment]
@KevinRoebert LGTM?

File: drasyl-core/src/main/java/org/drasyl/handler/dht/chord/ChordUtil.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    static {
+        OBJECT_MAPPER.addMixIn(IdentityPublicKey.class, LocalChordNode.IdentityPublicKeyMixin.class);
+        OBJECT_MAPPER.addMixIn(DrasylAddress.class, LocalChordNode.DrasylAddressMixin.class);
+    }
+
+    private static final long[] POWER_OF_TWO = new long[1 + Integer.SIZE];
+
+    static {
+        long value = 1;
+        for (int i = 0; i < POWER_OF_TWO.length; i++) {
```
Comment (KevinRoebert) [inline drasyl-core/src/main/java/org/drasyl/handler/dht/chord/ChordUtil.java:48]: If you really want to speed this up, you can also just enter fixed values here

File: drasyl-core/src/main/java/org/drasyl/handler/dht/chord/ChordFingerTable.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    public boolean hasSuccessor() {
+        return getSuccessor() != null;
+    }
+
+    public void removePeer(final DrasylAddress value) {
+        LOG.info("Remove peer `{}` from finger table.", value);
+        for (int i = Integer.SIZE; i > 0; i--) {
+            final DrasylAddress ithFinger = entries[i - 1];
+            if (ithFinger != null && ithFinger.equals(value)) {
```
Comment (KevinRoebert) [inline drasyl-core/src/main/java/org/drasyl/handler/dht/chord/ChordFingerTable.java:80]: if you can be sure, that `value` is not null, you could just do `value.equals(ithFinger)`

**Comment by KevinRoebert** [comment]
@HeikoBornholdt please add the magic numbers

https://thecodinglove.com/when-i-do-a-code-review-3

**Pull request merged**